### PR TITLE
vo_gpu_next: keep target params in priv instead of alocating it

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: video/out/vo.c [hunk 1]
```diff
@@ -597,6 +597,7 @@ static void run_reconfig(void *p)
     mp_mutex_lock(&vo->params_mutex);
     talloc_free(vo->params);
     vo->params = talloc_dup(vo, params);
+    vo->target_params = NULL;
```
Comment (sfan5) [inline video/out/vo.c:600]: what's the good reason for doing this?
this is the exact inverse of my reasoning in 2893b7d0f5

Comment (kasper93) [reply video/out/vo.c:600]: Ownership is on VO side, but why not make sure we don't leak some stray pointer after reconfig/deinit? I though https://github.com/mpv-player/mpv/commit/2893b7d0f51094bfee0cc27f3bebba8de0fd8018 was about memory ownership.

vo->target_params is a weak pointer that has to be initialized on each reconfig by the vo.

Comment (kasper93) [reply video/out/vo.c]: removed now

Comment (sfan5) [reply video/out/vo.c]: It was primarily about the memory ownership yes but I also don't see a reason to force a specific lifetime onto the VO impl.

**Review by sfan5 — APPROVED** [review]

**Comment by kasper93** [comment]
This is basically cosmetic change, let's get it out of the queue.

**Pull request merged**